### PR TITLE
chore: `__config` improved type hinting

### PR DESCRIPTION
**Pull request by thiskevinwang**
## Description

This PR improves the type-hinting for our global `__config` object

https://user-images.githubusercontent.com/26389321/228659027-14f134ff-1206-4a32-8f95-491dbb08ddf7.mp4

### CODE REVIEW

File: global.d.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ declare module '@hashicorp/react-search'
 /**
  * Application config, defined in environment-specific JSON files in `config/`
  */
-declare const __config: Record<string, any>
+declare const __config: typeof import('./config/base.json')
```
Comment (ashleemboyer) [inline global.d.ts:15]: Nice!! 🤩

File: src/types/products.ts [hunk 2]
```diff
@@ -20,7 +20,14 @@ type ProductName =
...
 
-type ProductSlug = Exclude<Products, 'hashicorp'> | 'hcp' | 'sentinel'
+type ProductSlug =
+	| Exclude<Products, 'hashicorp'>
+	| 'hcp'
+	| 'sentinel'
+	// This enables literals (e.g. 'boundary') in conjuction with a generic `string` type
+	// `(string & {})` is a workaround for this issue: https://github.com/Microsoft/TypeScript/issues/29729
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	| (string & {})
```
Comment (ashleemboyer) [inline src/types/products.ts:30]: 🙋🏻 Is this change directly related to the change in the other file? I'm not sure I understand how this could implicate our _many_ references to `ProductSlug`

Comment (thiskevinwang) [reply src/types/products.ts:30]: Yes, it's to accommodate for the improved typings introduced to `__config`, in `global.d.ts`.
That change ultimately replaced an `any` with `string`, which caused this line to have a type error:
- https://github.com/hashicorp/dev-portal/blob/kw.config-intellisense/src/components/command-bar/commands/search/components/integrations-tab-contents/index.tsx#L23

Comment (ashleemboyer) [reply src/types/products.ts:30]: Oh okay cool! That's helpful context. I'm going to check out the code locally real quick, that'll help me visualize these changes. Brb! 😊

Comment (ashleemboyer) [reply src/types/products.ts:30]: Have moved some discussion "offline" to Slack in case it's faster!

**Review by BRKalow — APPROVED** [review]
Works great! Nice improvement
